processes/consumer/flush.go

```diff
@@ -32,13 +32,20 @@ func Flush(ctx context.Context, inMemDB *models.DatabaseData, dest destination.B
 		return nil
 	}
 
-	var wg sync.WaitGroup
 	// Read lock to examine the map of tables
 	inMemDB.RLock()
 	allTables := inMemDB.TableData()
 	inMemDB.RUnlock()
 
-	// Flush will take everything in memory and call Snowflake to create temp tables.
+	if args.SpecificTable != "" {
+		if _, ok := allTables[args.SpecificTable]; !ok {
```

Comment:
Why do we need this?

Reply:
We're making making sure that we only flush tables that have the same name as `args.SpecificTable`, and that the data for said table isn't empty, but we don't have any checks that the table name that was passed in is actually in the in-memory database. This would only happen if we were to introduce a bug somewhere and pass in a bad value in for `args.SpecificTable`, so this is more of an illegal argument check than it is a check to prevent us from flushing the wrong thing since a bad value for `args.SpecificTable` is a no-op.